mshadow/packet-inl.h

```diff
@@ -324,22 +324,41 @@ template<typename E, PacketArch Arch>
 struct PacketCheck{
   static const bool kPass = false;
 };
-template<typename DType, PacketArch Arch>
-struct PacketCheck<ScalarExp<DType>, Arch> {
+template<PacketArch Arch>
+struct PacketCheck<ScalarExp<float>, Arch> {
   static const bool kPass = true;
 };
-template<int dim, typename DType, PacketArch Arch>
-struct PacketCheck<Tensor<cpu, dim, DType>, Arch> {
+template<int dim, PacketArch Arch>
+struct PacketCheck<Tensor<cpu, dim, float>, Arch> {
+  static const bool kPass = true;
+};
+template<typename OP, typename TA, int etype, PacketArch Arch>
+struct PacketCheck<UnaryMapExp<OP, TA, float, etype>, Arch> {
```

Comment:
No need to specialize the composites, only specialize the terminals should work(tensor, scalar)